src/main/java/org/sonarlint/intellij/core/ProjectServerNotifications.java

```diff
@@ -136,27 +142,39 @@ private NotificationConfiguration createConfiguration(SonarLintProjectSettings s
    * Simply displays the events and discards it
    */
   private class EventListener implements ServerNotificationListener {
-    @Override public void handle(ServerNotification notification) {
+    @Override
+    public void handle(ServerNotification notification) {
       SonarLintTelemetry telemetry = SonarLintUtils.getService(SonarLintTelemetry.class);
+      boolean isSonarCloud;
+      try {
+        ProjectBindingManager bindingManager = SonarLintUtils.getService(myProject, ProjectBindingManager.class);
+        ServerConnection server = bindingManager.getServerConnection();
+        isSonarCloud = server.isSonarCloud();
+      } catch (InvalidBindingException e) {
+        // do nothing
+        return;
+      }
```

Comment:
It seems a bit strange to me that we have to retrieve the connection from the notification, because we had it when we registered. Would be much simpler if we had notification.getServer() or notification.getConnection(). Or maybe keep the connection as a field of EventListener when we register

Reply:
I will have a look.